File: TDDMicroExercises/CSharp/TurnTicketDispenser/TicketDispenser.cs
namespace TDDMicroExercises.TurnTicketDispenser
{
    public class TicketDispenser
    {
        public TurnTicket GetTurnTicket()
        {
            int newTurnNumber = TurnNumberSequence.GetNextTurnNumber();
            TurnTicket newTurnTicket = new TurnTicket(newTurnNumber);

            return newTurnTicket;
        }
    }
}

File: Cpp/Source/TurnTicketDispenser/source/TicketDispenser.cpp

#include "../include/TicketDispenser.h"
#include "../include/TurnNumberSequence.h"

TurnTicket TicketDispenser::getTurnTicket()
{
    int newTurnNumber = TurnNumberSequence::getNextTurnNumber();
    return TurnTicket(newTurnNumber);
}
